lib/services/role_test.go

```diff
@@ -4693,7 +4964,7 @@ func TestCheckAccessToWindowsDesktop(t *testing.T) {
 			for i, check := range test.checks {
 				msg := fmt.Sprintf("check=%d, user=%v, server=%v, should_have_access=%v",
 					i, check.login, check.desktop.GetName(), check.hasAccess)
-				err := test.roleSet.checkAccess(check.desktop, test.state, NewWindowsLoginMatcher(check.login))
+				err := test.roleSet.checkAccess(check.desktop, AccessState{}, NewWindowsLoginMatcher(check.login))
```

Comment:
Is there a particular reason we're testing device verification works with server/db/kube but not desktop access?

Reply:
App access and desktop access run via the browser, where we don't have access to the low level device information needed for device trust.

Adding support for these browser-based workflows will be a research project for next quarter.